src/utils/texture.js

```diff
@@ -1,68 +1,206 @@
+/**
+ * Texture helpers for standard material component.
+ *
+ * @member textureCache {object} - Texture cache for:
+ *   - Images: textureCache has mapping of src -> repeat -> cached three.js texture.
+ *   - Videos: textureCache has mapping of videoElement -> cached three.js texture.
+ * @member videoCache {object} - Cache of video elements.
+ */
 var debug = require('./debug');
-var error = debug('components:texture:error');
 var THREE = require('../lib/three');
+
+var EVENTS = {
+  TEXTURE_LOADED: 'material-texture-loaded'
+};
+var error = debug('components:texture:error');
+var textureCache = {};
```

Comment:
Can move code to use systems (along with sound, light, camera code) after release.